test/nodes/test_web_retriever.py

```diff
@@ -1,163 +1,246 @@
-                meta={"title": "Eddard Stark", "link": "", "score": 0.9090909090909091},
-                id_hash_keys=["content"],
-                id="f408db6de8de0ffad0cb47cf8830dbb8",
-            ),
-            Document(
-                content="The most likely answer for the clue is NED. How many solutions does Arya Stark's Father have? With crossword-solver.io you will find 1 solutions. We use ...",
-                score=0.09090909090909091,
-                meta={
-                    "title": "Arya Stark's Father - Crossword Clue Answers",
-                    "link": "https://crossword-solver.io/clue/arya-stark%27s-father/",
-                    "position": 1,
-                    "score": 0.09090909090909091,
-                },
-                id_hash_keys=["content"],
-                id="51779277acf94cf90e7663db137c0732",
-            ),
-        ]
-    }
-
-    def mock_web_search_run(self, query: str) -> Tuple[Dict, str]:
-        return expected_search_results, "output_1"
-
-    class MockResponse:
-        def __init__(self, text, status_code):
-            self.text = text
-            self.status_code = status_code
-
-    def get(url, headers, timeout):
-        return MockResponse("mocked", 200)
-
-    def get_content(self, text: str) -> str:
-        return "What are the top solutions for\nArya Stark's Father\nWe found 1 solutions for\nArya Stark's Father\n.The top solutions is determined by popularity, ratings and frequency of searches. The most likely answer for the clue is NED..."
-
-    monkeypatch.setattr(WebSearch, "run", mock_web_search_run)
-    monkeypatch.setattr(ArticleExtractor, "get_content", get_content)
-    monkeypatch.setattr(requests, "get", get)
-
-    web_retriever = WebRetriever(api_key="", top_search_results=2, mode="preprocessed_documents")
-    result = web_retriever.retrieve(query="Who is the father of Arya Stark?")
-    assert len(result) == 1
-    assert isinstance(result[0], Document)
-    assert (
-        result[0].content
-        == "What are the top solutions for\nArya Stark's Father\nWe found 1 solutions for\nArya Stark's Father\n.The top solutions is determined by popularity, ratings and frequency of searches. The most likely answer for the clue is NED..."
+def test_scrape_links_with_search_results_with_preprocessor(mocked_requests, mocked_article_extractor):
+    wr = WebRetriever(api_key="fake_key", mode="preprocessed_documents")
+    preprocessor = PreProcessor(progress_bar=False)
+
+    sr1 = SearchResult("https://pagesix.com", "Some text", "0.43", "1")
+    sr2 = SearchResult("https://www.yahoo.com/", "Some text", "0.43", "2")
+    fake_search_results = [sr1, sr2]
+
+    result = wr._scrape_links(fake_search_results, "query", preprocessor)
+
+    assert isinstance(result, list)
+    assert all(isinstance(r, Document) for r in result)
+    # the documents from above SearchResult are so small that they will not be split into multiple documents
+    # by the preprocessor
```

Comment:
Then this test is almost equal to the previous...
Is it possible to provide longer Documents?

Reply:
You are right, but let's keep them both to ensure that PreProcessor's presence doesn't somehow mess things up. And yes, we can provide longer docs but then we'll be testing preprocessing which is moot if you ask me.